client/src/app/pages/applications/components/application-detail-drawer/application-facts.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  FilterCategory,
+  FilterToolbar,
+  FilterType,
+} from "@app/shared/components/FilterToolbar";
+import {
+  Button,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  ToolbarToggleGroup,
+} from "@patternfly/react-core";
+import React from "react";
+import FilterIcon from "@patternfly/react-icons/dist/esm/icons/filter-icon";
+import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+import { useTranslation } from "react-i18next";
+import { useLegacyFilterState } from "@app/shared/hooks/useLegacyFilterState";
+import { Fact } from "@app/api/models";
+import { FactDetailModal } from "./fact-detail-modal/fact-detail-modal";
+
+export interface IApplicationRiskProps {
+  facts: Fact[];
+}
+
+export const ApplicationFacts: React.FC<IApplicationRiskProps> = ({
+  facts,
+}) => {
+  const { t } = useTranslation();
+  const sources = new Set<string>();
+
+  // TODO: work through how to store sources for facts in the ui for sorting
+  //   facts.forEach((fact) => sources.add(fact.source || ""));
+
+  const filterCategories: FilterCategory<Fact, "source">[] = [
+    {
+      key: "source",
+      title: t("terms.source"),
+      type: FilterType.multiselect,
+      placeholderText: t("terms.source"),
+      //   getItemValue: (fact) => fact.source || "Source default name",
+      selectOptions: Array.from(sources)
+        //TODO: Custom sorting for facts may be required
+        // .sort(compareSources)
+        .map((source) => source || "Source default name")
+        .map((source) => ({ key: source, value: source })),
+      logicOperator: "OR",
+    },
+  ];
+
+  const {
+    filterValues,
+    setFilterValues,
+    filteredItems: filteredFacts,
+  } = useLegacyFilterState(facts, filterCategories);
+
+  const [selectedFactForDetailModal, setSelectedFactForDetailModal] =
+    React.useState<Fact | null>(null);
+
+  return (
+    <>
+      <Toolbar
+        clearAllFilters={() => setFilterValues({})}
+        clearFiltersButtonText={t("actions.clearAllFilters")}
+      >
+        <ToolbarContent className={spacing.p_0}>
+          <ToolbarItem>Filter by:</ToolbarItem>
+          <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">
+            <FilterToolbar
+              filterCategories={filterCategories}
+              filterValues={filterValues}
+              setFilterValues={setFilterValues}
+              showFiltersSideBySide
+            />
+          </ToolbarToggleGroup>
+        </ToolbarContent>
+      </Toolbar>
+      {filteredFacts.map((fact) => {
+        return (
+          <div>
+            <Button
```

Comment:
This flat list of link-buttons with no real layout aside from line breaks really bothers me for some reason, but that's a design issue. Maybe we can revisit this with Justin. I think even just a compact table with one unlabeled column, or maybe a `DataList` (https://www.patternfly.org/v4/components/data-list), something to make it look less plain. Outside the scope of this PR though, just thinking out loud.

Reply:
Opened an issue for this & linked in the description.